cpp/src/detail/shuffle_vertex_pairs.cu

```diff
@@ -519,4 +519,69 @@ shuffle_int_vertex_pairs_with_values_to_local_gpu_by_edge_partitioning(
   std::vector<int64_t> const& vertex_partition_range_lasts);
 
 }  // namespace detail
+
```

Comment:
It might make sense to move this file out of `src/detail`.  Everything in this directory is in the detail namespace, if we're exposing it to the public namespace then we might want to move the source file to `src/utilities`.